### PR TITLE
replace some constants with configurable properties for raycast-vehicle

### PR DESCRIPTION
**Pull request by tomo0613**
Greetings!

I'd like to use some constants from the RaycastVehicle class [especially fwdFactor] as configurable properties, in the WheelInfo class.
Tweaking that value makes it possible to achieve more realistic handling for a raycast-vehicle.

I really like the vehicle behaviour in [Sketchbook](https://github.com/swift502/Sketchbook), and I want to improve [my project](https://github.com/tomo0613/offroadJS_v2_demo) as well without using a custom cannon lib.
I also slightly modified the example, imho the [original demo](http://schteppe.github.io/cannon.js/demos/raycastVehicle.html) feels a bit weird when braking [press B] (but it could be reverted)

### CODE REVIEW

File: src/objects/WheelInfo.ts [hunk 1]
```diff
@@ -128,6 +132,8 @@ export class WheelInfo {
       dampingCompression: 10,
       dampingRelaxation: 10,
       frictionSlip: 10000,
```
Comment (swift502) [inline src/objects/WheelInfo.ts:134]: Wouldn't it also make sense to make this value more realistic by default? Something in the range of 0.5 - 2?

**Comment by tomo0613** [comment]
It would, absolutely, I just didn't want to alter the current behaviour,
but maybe that's not a big deal.

swift502 <notifications@github.com> ezt írta (időpont: 2021. jan. 13., Sze,
13:10):

**Comment by swift502** [comment]
@tomo0613 yeah sorry I deleted that comment because I realised it's something different than just exposing parameters. Maybe that's for a different PR.

Your current changes should (I believe) definitely be merged though.

**Comment by tomo0613** [comment]
Thanks a lot 👍

swift502 <notifications@github.com> ezt írta (időpont: 2021. jan. 13., Sze
21:25):

**Comment by marcofugaro** [comment]
@tomo0613 Nice! I agree they would make the vehicle configuration better. We should expose them.

I don't think we should expose the variables this way though. [They come from bullet](https://github.com/bulletphysics/bullet3/blob/df09fd9ed37e365ceae884ca7f620b61607dae2e/src/BulletDynamics/Vehicle/btRaycastVehicle.cpp#L547-L548), and probably there is a reason for them to have that value.

Instead we should do it like this: 

```js
const x = wheel.forwardImpulse * fwdFactor / wheel.forwardAcceleration
const y = wheel.sideImpulse * sideFactor / wheel.sideAcceleration
```

`forwardAcceleration` and `sideAcceleration` should both default to `1`. There is a division because that way, if the value is greater, the car goes faster in the forward direction. The same is for the `sideAcceleration`.

Could you update your PR with this?

> I also slightly modified the example, imho the original demo feels a bit weird when braking [press B] (but it could be reverted)

True, it actually feels better 👍, keep the friction at 1.4. Don't change the `forwardAcceleration` and `sideAcceleration` though, leave them as default.

**Comment by marcofugaro** [comment]
@swift502 what was your proposition?

The new version of Sketchbook looks nice btw! Did you notice we integrated one of your original PR to cannon in 2970232ccbf0757db712d369cb84240f8deee0da?

**Comment by tomo0613** [comment]
I have updated the branch based on the suggestions.

**Comment by swift502** [comment]
@marcofugaro I proposed changing the default frictionSlip value to something smaller (In Sketchbook I use 0.8, and I gather @tomo0613 uses 1.4?). A value of 10000 seems excessivelly unrealistic and I believe it's what makes a lot of vehicle demos online feel very unnatural. I don't really know why @schteppe made 10000 the default.

As for 2970232ccbf0757db712d369cb84240f8deee0da, yeah I noticed. 🙂 I planned on switching from cannon to cannon-es in Sketchbook, but never got around to it for some reason.

**Comment by marcofugaro** [comment]
@swift502 probably an oversight, it is 10.5 in bullet by default.

https://github.com/bulletphysics/bullet3/blob/afa4fb54505fd071103b8e2e8793c38fd40f6fb6/src/BulletDynamics/Vehicle/btRaycastVehicle.h#L45

Thanks @tomo0613, will merge it and publish tonight

**Comment by swift502** [comment]
@marcofugaro Good idea to compare with Bullet. Even 10.5 would imo be a better default value.

**Review by marcofugaro — APPROVED** [review]

**Comment by tomo0613** [comment]
I'm happy to see the changes in live, thank you all!

**Pull request merged**